### PR TITLE
DEVEXP-337: Mocked tests setup

### PR DESCRIPTION
**Pull request by asein-sinch**
- Cucumber configuration
 - Steps implementation for Fax.Faxes domain as a POC
 - CI configuration to put the feature files in the right folder and run the cucumber tests

### CODE REVIEW

File: packages/fax/tests/rest/v3/faxes/faxes.steps.ts [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+let listResponse: PageResult<Fax.Fax>;
+const faxList: Fax.Fax[] = [];
+let sendFaxResponse: Fax.Fax[];
+let fax: Fax.Fax;
+let fileBuffer: FileBuffer;
+let deleteContentResponse: void;
+
+Given('the Fax service is available', function () {
+  faxService = new FaxService({
+    projectId: 'tinyfrog-jump-high-over-lilypadbasin',
```
Comment (JPPortier) [inline packages/fax/tests/rest/v3/faxes/faxes.steps.ts:16]: 🐸

File: .github/workflows/run-ci.yaml [hunk 2]
```diff
@@ -20,6 +20,50 @@ jobs:
...
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Checkout sinch-sdk-mockserver repository
+        uses: actions/checkout@v3
+        with:
+          repository: sinch/sinch-sdk-mockserver
+          ref: DEVEXP-337_Mockserver-POC
+          token: ${{ secrets.PAT_CI }}
```
[the remark this answers is not in the document]
Comment (asein-sinch) [reply .github/workflows/run-ci.yaml:33]: Yes, this is a personal access token in this case which grants me access to the private repo from the Sinch organization.

File: .github/workflows/run-ci.yaml [hunk 3]
```diff
@@ -20,6 +20,50 @@ jobs:
...
+          while ! docker inspect --format='{{json .State.Health.Status}}' $(docker-compose ps -q fax-server) | grep -q '"healthy"'; do
+            echo "Waiting for fax-server to be healthy..."
+            sleep 2
+          done
+      - name: Create target directories for feature files
+        run: |
+          mkdir -p ./packages/fax/tests/e2e/features
+      - name: Copy feature files
+        run: |
+          cp sinch-sdk-mockserver/features/fax/*.feature ./packages/fax/tests/e2e/features/
```
[the remark this answers is not in the document]
Comment (asein-sinch) [reply .github/workflows/run-ci.yaml:60]: The destination folder has to match the Cucumber configuration. If we say to Cucumber to read from the docker volume, that's fine too. As we don't have an artifact repositoty to rely on for the moment, cloning the docker-compose file and the features is the simplest solution that came into my mind in a spirit of sharing the mockserver expectations and cucumber features across the SDKs

File: .github/workflows/run-ci.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (JPPortier) [reply .github/workflows/run-ci.yaml:33]: Thank you for confirmation

**Review by JPPortier — APPROVED** [review]

**Pull request merged**